src/main/java/org/opentripplanner/middleware/controllers/api/ItineraryCheckController.java

```diff
@@ -0,0 +1,120 @@
+package org.opentripplanner.middleware.controllers.api;
+
+import com.beerboy.ss.SparkSwagger;
+import com.beerboy.ss.rest.Endpoint;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.eclipse.jetty.http.HttpStatus;
+import org.opentripplanner.middleware.models.ItineraryExistenceResult;
+import org.opentripplanner.middleware.models.MonitoredTrip;
+import org.opentripplanner.middleware.otp.response.Itinerary;
+import org.opentripplanner.middleware.otp.response.OtpResponse;
+import org.opentripplanner.middleware.utils.DateTimeUtils;
+import org.opentripplanner.middleware.utils.HttpUtils;
+import org.opentripplanner.middleware.utils.ItineraryUtils;
+import org.opentripplanner.middleware.utils.JsonUtils;
+import spark.Request;
+import spark.Response;
+
+import java.net.URISyntaxException;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Map;
+
+import static com.beerboy.ss.descriptor.EndpointDescriptor.endpointPath;
+import static com.beerboy.ss.descriptor.MethodDescriptor.path;
+import static org.opentripplanner.middleware.utils.HttpUtils.JSON_ONLY;
+import static org.opentripplanner.middleware.utils.JsonUtils.getPOJOFromRequestBody;
+import static org.opentripplanner.middleware.utils.JsonUtils.logMessageAndHalt;
+
+/**
+ * This endpoint checks is used by web clients or other APIs to let users know
+ * whether a given itinerary exists on each day of the week.
+ */
+public class ItineraryCheckController implements Endpoint {
```

Comment:
This design seems a little clunky to me. This entire controller does not need to exist. Here's what I propose:
1. In the UI, when a user clicks save trip, we send a request to createMonitoredTrip.
2. As part of the creation of that monitored trip, we kick off a checkItinerary job (Runnable), that gets tracked in a status object as part of the MonitoredTrip it sends back in the create trip response.
3. Meanwhile in the UI, we process the response from create trip and keep polling the get monitored trip endpoint to check the check itinerary status until it is complete. As itinerary existence is confirmed on the various days, that status gets updated.
4. Once the check is complete, the monitored trip status is fully updated and the user can proceed with confirming the trip/notification details.

Reply:
Oh, actually the check already happens in `initializeTripAndSetVerifiedItinerary`. I need to dig into this a little more to understand why the separate controller exists then.

Reply:
This controller is for the UI to show which days are possible or not.